### ISSUE TITLE
Error loop: Error: browserController.newPage() failed: Cause:browserController.newPage() timed out

### ISSUE THREAD

**Issue by beshoo** [drobnikj/crawler-google-places#193]
Crawler manage to crawl 111 addresses and start looking at this error 
Am monitoring the proxy network. "crawler-google-places" does not send any request but keeps looping with the previous error.
```
autoscaled_pool.js:408:17)
INFO  [STATS]: failed: 0 | ok: 114 | outOfPolygon: 858 | outOfPolygonCached: 858 | places: 107 | maps: 7
INFO  Statistics: PuppeteerCrawler request statistics: 

{"requestAvgFailedDurationMillis":null,"requestAvgFinishedDurationMillis":13188,"requestsFinishedPerMinute":6,"requestsFailedPerMinute":0,"requestTotalDurationMillis":1503420,"requestsTotal":114,"crawlerRuntimeMillis":1199916,"retryHistogram":[114]}
INFO  PuppeteerCrawler:AutoscaledPool: state {"currentConcurrency":1,"desiredConcurrency":1,"systemStatus":{"isSystemIdle":false,"memInfo":{"isOverloaded":false,"limitRatio":0.2,"actualRatio":0},"eventLoopInfo":{"isOverloaded":false,"limitRatio":0.6,"actualRatio":0.04},"cpuInfo":{"isOverloaded":true,"limitRatio":0.4,"actualRatio":0.914},"clientInfo":{"isOverloaded":false,"limitRatio":0.3,"actualRatio":0}}}
ERROR PuppeteerCrawler: handleRequestFunction failed, reclaiming failed request back to the list or queue {"url":"https://www.google.com/maps/@33.47905712574731,36.30314359075436,17z/search","retryCount":2,"id":"s8Zi0cDnrxrguKK"}
  Error: browserController.newPage() failed: vMSTtUnFXDYGi5F4MFE15
  Cause:browserController.newPage() timed out..
      at BrowserPool._createPageForBrowser (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/browser-pool/dist/browser-pool.js:326:19)
      at async PuppeteerCrawler._handleRequestFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/browser_crawler.js:306:22)
      at async PuppeteerCrawler._runTaskFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/basic_crawler.js:425:13)
      at async AutoscaledPool._maybeRunTask (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/autoscaling/autoscaled_pool.js:408:17)

ERROR PuppeteerCrawler: handleRequestFunction failed, reclaiming failed request back to the list or queue {"url":"https://www.google.com/maps/@33.47189662590464,36.29885171792391,17z/search","retryCount":2,"id":"mQrkToxPRQDXyNe"}
  Error: browserController.newPage() failed: 4BfK73uIrfFTPxct6aBp_
  Cause:browserController.newPage() timed out..
      at BrowserPool._createPageForBrowser (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/browser-pool/dist/browser-pool.js:326:19)
      at async PuppeteerCrawler._handleRequestFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/browser_crawler.js:306:22)
      at async PuppeteerCrawler._runTaskFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/basic_crawler.js:425:13)
      at async AutoscaledPool._maybeRunTask (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/autoscaling/autoscaled_pool.js:408:17)

```

input:
```

{
  searchStringsArray: [ 'بقالية' ],
  country: 'Syria',
  city: 'Damascus',
  county: '',
  postalCode:'',
  maxCrawledPlaces: 99999,
  maxCrawledPlacesPerSearch: 1000,
  maxReviews: 20,
  language: 'en',
  maxImages: 1,
  includeHistogram: true,
  includeOpeningHours: true,
  includePeopleAlsoSearch: false,
  additionalInfo: false,
  reviewsSort: 'mostRelevant',
  reviewsTranslation: 'onlyTranslated',
  exportPlaceUrls: false,
  zoom: 17,
  maxAutomaticZoomOut: 13,
  
  proxyConfig: {
    useApifyProxy: false,
  },
      maxConcurrency: 20,
      maxPageRetries: 2,
      pageLoadTimeoutSec: 60,
      maxPagesPerBrowser: 10,
      useChrome: false,
      useStealth: false,
      scrapeReviewerName: true,
      scrapeReviewerId: true,
      scrapeReviewerUrl: true,
      scrapeReviewId: true,
      scrapeReviewUrl: true,
      scrapeReviewerName:true,
      scrapeResponseFromOwnerText: true,
      debug:false,
};

```

**Comment by beshoo** [comment]
Killing the processes and restart it does not solve the problem

```
INFO  System info {"apifyVersion":"1.3.4","apifyClientVersion":"1.4.2","osType":"Linux","nodeVersion":"v14.17.6"}
WARN  The following Key-value store directory contains a previous state:
      /home/beshoo/www/google-map/dataset/key_value_stores/default
      If you did not intend to persist the state - please clear the respective directory and re-start the actor.
WARN  The following Request queue directory contains a previous state:
      /home/beshoo/www/google-map/dataset/request_queues/default
      If you did not intend to persist the state - please clear the respective directory and re-start the actor.
INFO  [Geolocation]: Location found: Damascus, Damascus Governorate, 2277, Syria, lat: 33.5130695, long: 36.3095814
WARN  Actor was restarted, skipping search step because it was already done...
INFO  PuppeteerCrawler:AutoscaledPool:Snapshotter: Setting max memory of this run to 32184 MB. Use the APIFY_MEMORY_MBYTES environment variable to override it.
INFO  PuppeteerCrawler:AutoscaledPool: state {"currentConcurrency":0,"desiredConcurrency":2,"systemStatus":{"isSystemIdle":true,"memInfo":{"isOverloaded":false,"limitRatio":0.2,"actualRatio":null},"eventLoopInfo":{"isOverloaded":false,"limitRatio":0.6,"actualRatio":null},"cpuInfo":{"isOverloaded":false,"limitRatio":0.4,"actualRatio":null},"clientInfo":{"isOverloaded":false,"limitRatio":0.3,"actualRatio":null}}}
ERROR PuppeteerCrawler: handleRequestFunction failed, reclaiming failed request back to the list or queue {"url":"https://www.google.com/maps/@33.55782262401667,36.31172733641526,17z/search","retryCount":1,"id":"G862phs9l1XAePm"}
  Error: browserController.newPage() failed: 2WgLzj20_RoxEYoOu_sk3
  Cause:browserController.newPage() timed out..
      at BrowserPool._createPageForBrowser (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/browser-pool/dist/browser-pool.js:326:19)
      at async PuppeteerCrawler._handleRequestFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/browser_crawler.js:306:22)
      at async PuppeteerCrawler._runTaskFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/basic_crawler.js:425:13)
      at async AutoscaledPool._maybeRunTask (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/autoscaling/autoscaled_pool.js:408:17)
ERROR PuppeteerCrawler: handleRequestFunction failed, reclaiming failed request back to the list or queue {"url":"https://www.google.com/maps/@33.55424237409534,36.31172733641526,17z/search","retryCount":1,"id":"eVOj0KtQIH89Ac3"}
  Error: browserController.newPage() failed: 2WgLzj20_RoxEYoOu_sk3
  Cause:browserController.newPage() timed out..
      at BrowserPool._createPageForBrowser (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/browser-pool/dist/browser-pool.js:326:19)
      at async PuppeteerCrawler._handleRequestFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/browser_crawler.js:306:22)
      at async PuppeteerCrawler._runTaskFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/basic_crawler.js:425:13)
      at async AutoscaledPool._maybeRunTask (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/autoscaling/autoscaled_pool.js:408:17)

```

**Comment by beshoo** [comment]
I note a lot of processes in my Linux top list

` \_ /home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/puppeteer/.local-chromium/linux-901912/chrome-linux/chrome --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=Translate --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --disable-sync --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --enable-blink-features=IdleDetection --user-data-dir=/exe_tmp/PFSPWLRjHs8xn7-hlrBZS --headless --hide-scrollbars --mute-audio about:blank --disable-web-security --no-sandbox --disable-features=IsolateOrigins,site-per-process --lang=en --user-agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.67 Safari/537.36 --disable-blink-features=AutomationControlled --proxy-server=http://127.0.0.1:43379 --remote-debugging-port=0
`
I tried to kill all of them but it seems to keep appearing while the crawler is **OFF** ...

**Comment by metalwarrior665** [comment]
Hello, your CPU probably stopped keeping up. Try to reduce maxConcurrency.

**Comment by beshoo** [comment]
@metalwarrior665 the problem is when you killed the node process, the Chrome process keep looping in the background. Even if you kill -9 the Chrome process, another process will be started.

What is the best practices for this repo if you wanto to run more than 50 search in parallel?

Regard

**Comment by metalwarrior665** [comment]
You mean the one `process.exit` that is in the code? Please keep in mind this whole repo is optimized to run well on Apify platform. Locally, everyone has to figure out how to optimally run it on their hardware.

**Comment by beshoo** [comment]
I mean when you ctrl+c the node process. Why it dos not end related chrome
processes?


Regarding optimize thing, i understand it is ment to be used with apify
platform, but aye the end of the day, it is node application.

Thanks for the help.

**Comment by beshoo** [comment]
@metalwarrior665 

The strange problem I have, that I can not even start the new search
once I start the search I got

```
INFO  System info {"apifyVersion":"1.3.4","apifyClientVersion":"1.4.2","osType":"Linux","nodeVersion":"v14.17.6"}
WARN  The following Key-value store directory contains a previous state:
      /home/beshoo/www/google-map/dataset/key_value_stores/default
      If you did not intend to persist the state - please clear the respective directory and re-start the actor.
WARN  The following Request queue directory contains a previous state:
      /home/beshoo/www/google-map/dataset/request_queues/default
      If you did not intend to persist the state - please clear the respective directory and re-start the actor.
INFO  [Geolocation]: Location found: Damascus, Damascus Governorate, 2277, Syria, lat: 33.5130695, long: 36.3095814
WARN  Actor was restarted, skipping search step because it was already done...
INFO  PuppeteerCrawler:AutoscaledPool:Snapshotter: Setting max memory of this run to 32184 MB. Use the APIFY_MEMORY_MBYTES environment variable to override it.
INFO  PuppeteerCrawler:AutoscaledPool: state {"currentConcurrency":0,"desiredConcurrency":2,"systemStatus":{"isSystemIdle":true,"memInfo":{"isOverloaded":false,"limitRatio":0.2,"actualRatio":null},"eventLoopInfo":{"isOverloaded":false,"limitRatio":0.6,"actualRatio":null},"cpuInfo":{"isOverloaded":false,"limitRatio":0.4,"actualRatio":null},"clientInfo":{"isOverloaded":false,"limitRatio":0.3,"actualRatio":null}}}
ERROR PuppeteerCrawler: handleRequestFunction failed, reclaiming failed request back to the list or queue {"url":"https://www.google.com/maps/@33.522020124803326,36.26022486244986,17z/search","retryCount":1,"id":"Q6ZYTebValcCIqm"}
  Error: browserController.newPage() failed: psb6N3mOoXVly4ALATZzZ
  Cause:browserController.newPage() timed out..
      at BrowserPool._createPageForBrowser (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/browser-pool/dist/browser-pool.js:326:19)
      at async PuppeteerCrawler._handleRequestFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/browser_crawler.js:306:22)
      at async PuppeteerCrawler._runTaskFunction (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/crawlers/basic_crawler.js:425:13)
      at async AutoscaledPool._maybeRunTask (/home/beshoo/public_html/google-map/crawler-google-places-master/node_modules/apify/build/autoscaling/autoscaled_pool.js:408:17)
ERROR PuppeteerCrawler: handleRequestFunction failed, reclaiming failed request back to the list or queue {"url":"https://www.google.com/maps/@33.51843987488199,36.26022486244986,17z/search","retryCount":1,"id":"FA6cvJnNyBTd8QY"}

```

I cleared the dataset, redownload the project ...something!
Please advice

**Comment by metalwarrior665** [comment]
As I said, we don't have capacity to test on various operating systems and machines etc. It works well on Apify platform and for most other people using it locally and that is good enough. I usually get this error when I do some other stuff on the computer so there is not enough resources.

**Comment by beshoo** [comment]
This problem is related to 2 things.
OS and maxConcurrency
When you use Centos, you will see a lot of problems jumping around. I advise users to use the docker version.
maxConcurrency, yes set this between 2 and 5.
